src/pytest_inmanta_lsm/lsm_project.py

```diff
@@ -691,39 +739,20 @@ def auto_transfer(self, service_id: uuid.UUID) -> inmanta_lsm.model.ServiceInsta
             transfer_type=inmanta_lsm.const.TransferTrigger.AUTO,
         )
 
-        def next_state(state: str) -> None:
-            """
-            Apply this state to our service, if it is different from the current
-            state, also increment the version
-
-            :param state: The new state to apply
-            """
-            if service.state == state:
-                return
-
-            service.last_updated = datetime.datetime.now()
-            service.version += 1
-            service.state = state
-
         try:
-            # Trigger a compile for the transition
             LOGGER.info(
                 "Triggering compile on state %s before auto transfer (%s) for service %s (%s)",
                 service.state,
                 transfer.description,
                 service.id,
                 service.service_entity,
             )
-            self.compile(
-                service_id=service.id,
-                validation=transfer.validate_,
```

Comment:
But the logic here was wrong (at least for the update path), we were compiling for the current state, but with the next transfer, it should have been the transfer we just took right?